### PR TITLE
Implement user.exec auth support

### PR DESCRIPTION
**Pull request by jakolehm**
PR implements support for:

```yaml
users:
- name: my-user
  user:
    exec:
      # Command to execute. Required.
      command: "example-client-go-exec-plugin"

      # API version to use when decoding the ExecCredentials resource. Required.
      #
      # The API version returned by the plugin MUST match the version listed here.
      #
      # To integrate with tools that support multiple versions (such as client.authentication.k8s.io/v1alpha1),
      # set an environment variable or pass an argument to the tool that indicates which version the exec plugin expects.
      apiVersion: "client.authentication.k8s.io/v1beta1"

      # Environment variables to set when executing the plugin. Optional.
      env:
      - name: "FOO"
        value: "bar"

      # Arguments to pass when executing the plugin. Optional.
      args:
      - "arg1"
      - "arg2"
```

### CODE REVIEW

File: lib/k8s/transport.rb [hunk 1]
```diff
@@ -75,21 +75,46 @@ def self.config(config, server: nil, **overrides)
...
+    end
+
+    # @param exec_conf [K8s::Config::UserExec]
+    # @return [String]
+    def self.token_from_exec(exec_conf)
+      cmd = [exec_conf.command]
+      cmd = cmd + exec_conf.args if exec_conf.args
+      orig_env = ENV.to_h
+      if envs = exec_conf.env
+        envs.each do |env|
```
Comment (kke) [inline lib/k8s/transport.rb]: `ENV.update(hash)` should do this in one line

Comment (jakolehm) [reply lib/k8s/transport.rb]: Idk if it makes this more easy to read, `envs` is an array of hashes?

Comment (jnummelin) [reply lib/k8s/transport.rb]: also one option is to put those envs into the command to be used, that way there's no need to "manage" the current env at all

**Comment by kke** [comment]
Seems a bit dangerous, maybe there could be an option to enable?

**Comment by jakolehm** [comment]
> Seems a bit dangerous, maybe there could be an option to enable?

Wanna explain more? 😄

**Comment by kke** [comment]
> Wanna explain more? 😄 This is part of the kubeconfig spec, nothing we can really do here (other than support it).

If someone uses k8s-client to implement something that takes a config, it may be a surprise that the client executes arbitary commands from that config, maybe not, maybe mention in yardoc/readme.

**Pull request merged**